src/FileWriterTask.cpp

```diff
@@ -25,119 +25,120 @@ json hdf_parse(std::string const &Structure) {
-  }
+  // Add new source
+  Demuxers.emplace_back(Source.topic());
+  auto &d = Demuxers.back();
+  d.add_source(std::move(Source));
+}
 
+void FileWriterTask::InitialiseHdf(std::string const &NexusStructure,
+                                   std::string const &ConfigFile,
+                                   std::vector<StreamHDFInfo> &HdfInfo,
+                                   bool UseSwmr) {
   auto NexusStructureJson = hdf_parse(NexusStructure);
   auto ConfigFileJson = hdf_parse(ConfigFile);
 
   try {
-    LOG(Sev::Info, "Creating HDF file {}", filename_full);
-    hdf_file.init(filename_full, NexusStructureJson, ConfigFileJson,
-                  stream_hdf_info, UseHDFSWMR);
-  } catch (...) {
-    std::throw_with_nested(std::runtime_error(fmt::format(
-        "can not initialize hdf file  hdf_output_prefix: {}  hdf_filename: {}",
-        hdf_output_prefix, hdf_filename)));
+    LOG(Sev::Info, "Creating HDF file {}", Filename);
+    hdfFile.init(Filename, NexusStructureJson, ConfigFileJson, HdfInfo,
+                 UseSwmr);
+    // The HDF file is closed and re-opened to (optionally) support SWMR and
+    // parallel writing.
+    closeFile();
+    reopenFile();
+
+  } catch (std::exception const &E) {
+    std::throw_with_nested(std::runtime_error(
+        fmt::format("can not initialize hdf file {}", Filename)));
   }
 }
 
-void FileWriterTask::hdf_close_before_reopen() { hdf_file.close(); }
+void FileWriterTask::closeFile() { hdfFile.close(); }
 
-int FileWriterTask::hdf_reopen() {
+void FileWriterTask::reopenFile() {
   try {
-    hdf_file.reopen(filename_full, json::object());
+    hdfFile.reopen(Filename, json::object());
   } catch (std::exception const &E) {
     LOG(Sev::Error, "Exception: {}", E.what());
     if (StatusProducer) {
-      logEvent(StatusProducer, StatusCode::Error, ServiceID, JobID,
+      logEvent(StatusProducer, StatusCode::Error, ServiceId, JobId,
                fmt::format("Exception: {}", E.what()));
     }
-    return -1;
+    throw;
   }
-  return 0;
 }
 
-/// \brief  The unique numeric identifier of this job. Could maybe be replaced
-/// by `JobID`.
-uint64_t FileWriterTask::id() const { return _id; }
+uint64_t FileWriterTask::id() const { return Id; }
+
+std::string FileWriterTask::jobID() const { return JobId; }
+
+hdf5::node::Group FileWriterTask::hdfGroup() { return hdfFile.H5File.root(); }
 
-/// \brief  The `JobID` of this file being written.
-std::string FileWriterTask::job_id() const { return JobID; }
+bool FileWriterTask::swmrEnabled() const { return hdfFile.isSWMREnabled(); }
 
-/// \brief  Set our `JobID` to the passed value.
-void FileWriterTask::job_id_init(std::string const &JobID_) { JobID = JobID_; }
+void FileWriterTask::setJobId(std::string const &Id) { JobId = Id; }
 
-/// \brief  Return statistics about this job as JSON.
 json FileWriterTask::stats() const {
   auto Topics = json::object();
-  for (auto &d : _demuxers) {
+  for (auto &d : Demuxers) {
```

Comment:
Single character variable.
```suggestion
  for (auto &CurrentDemuxer : Demuxers) {
```